### ISSUE TITLE
Do not always jump to max if there are bounds and it's "closer" to min

### ISSUE THREAD

**Issue by vdsh** [andrejewski/angle-input#2]
Thanks for the code, really helpful.

However, something bugged me when I put a max to 180. Whenever I got close to 0 but like 356°, it jumped to my max (where I think it makes more sense to have it at 0 (as it's "closer")

So this is what I added at the very beginning of the normalize function, but I am not sure whether it's the best way to proceed

function normalize(degree) {
      // if outside bounds, but closer to min, returns min
      if (degree > max) {
        if (Math.abs(degree - max) > Math.abs(degree - 360 - min)) {
          return min;
        }
        return max;
      }

**Comment by andrejewski** [comment]
I think you're making the right assessment UX wise. If the bounds are say [0,180] then clicking around [270-360] ought to bring you to 0 as this is more likely what the user was meaning to choose. 

If your code works for you that's great! If we wanted to merge a change back into this repo we'd need to factor in the `step`, etc so it all worked together.

**Comment by vdsh** [comment]
Thanks for the quick reply. Why do we need the step? As it's either min or max, there is no impact from the step, right?

**Comment by andrejewski** [comment]
Say for example the step was 90deg, I think we'd want the same "go to nearest" behavior if someone clicked >45deg & <135deg

**Comment by vdsh** [comment]
Indeed... I did not think about that... I will think of a solution, it will probably be somewhat similar. but I have a bit a hard time understanding the remaining of the function
